fix(values): consistent usage of postgresql.adminPasswordKey

muja opened the pull request:
<!--
Thank you for your contribution! Complete the following fields to provide insight into the changes being requested as well as steps that you can take to ensure it meets all of the requirements

Please remember to:
- mention any issue(s) that this PR closes using a closing keyword as well as the issue number, such as "Closes #XYZ" or "Resolves janus-idp/repo-name#XYZ", cf.
  [documentation](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
- ensure there are no merge commits!

 -->

## Description of the change

Consistently use adminPasswordKey. Overriding this value currently leads to buggy behavior because it is not consistently used everywhere. Specifically the hard-coded `postgres-password` in one of the `extraEnvVars` is causing issues and downstream values files cannot override that easily.

## Additional Information

Since `janus-idp.postgresql.secretName` is not used anymore, could be considered to delete it, but it might break downstream charts which use that helper.

## Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the `values.yaml` and added to the README.md. The [pre-commit](https://pre-commit.com/) utility can be used to generate the necessary content. Use `pre-commit run -a` to apply changes.
- [x] JSON Schema template updated and re-generated the raw schema via `pre-commit` hook.
- [x] List tests pass for Chart using the [Chart Testing](https://github.com/helm/chart-testing) tool and the `ct lint` command.

muja committed:
fix(values): consistent usage of postgresql.adminPasswordKey

rm3l reviewed (commented):
Hi @muja ! Thanks for this contribution.

```
Error:  templates/: template: backstage/charts/upstream/charts/postgresql/templates/primary/statefulset.yaml:385:16: executing "backstage/charts/upstream/charts/postgresql/templates/primary/statefulset.yaml" at <include "common.tplvalues.render" (dict "value" .Values.primary.extraEnvVars "context" $)>: error calling include: template: backstage/charts/common/templates/_tplvalues.tpl:19:8: executing "common.tplvalues.render" at <tpl $value .context>: error calling tpl: error during tpl function execution for "- name: POSTGRESQL_ADMIN_PASSWORD\n  valueFrom:\n    secretKeyRef:\n      key: '{{- include \"postgresql.adminPasswordKey\" }}'\n      name: '{{- include \"postgresql.v1.secretName\" . }}'": template: backstage/charts/upstream/charts/postgresql/templates/primary/statefulset.yaml:4:16: executing "backstage/charts/upstream/charts/postgresql/templates/primary/statefulset.yaml" at <include>: wrong number of args for include: want 2 got 1
```

Could you look into the lint issues reported?
https://github.com/redhat-developer/rhdh-chart/actions/runs/10955059451/job/30920518609?pr=44
Thanks.

muja commented:
Hi @rm3l 

I believe I was missing a `.`. Should be fixed now.

muja commented:
It seems that the context differs for the two invocations. The invocations on line 111/112 happen from the `upstream` chart, where the `.Values.global` look like this (weirdly):

```json
{
    "auth": {
        "backend": {
            "enabled": true,
            "existingSecret": "",
            "value": ""
        }
    },
    "clusterRouterBase": "apps.example.com",
    "dynamic": {
        "includes": [
            "dynamic-plugins.default.yaml"
        ],
        "plugins": []
    },
    "host": "",
    "imagePullSecrets": [],
    "imageRegistry": ""
}
```

while from the `postgres` chart's POV they look like this:

```json
{
    "auth": {
        "backend": {
            "enabled": true,
            "existingSecret": "",
            "value": ""
        }
    },
    "clusterRouterBase": "apps.example.com",
    "dynamic": {
        "includes": [
            "dynamic-plugins.default.yaml"
        ],
        "plugins": []
    },
    "host": "",
    "imagePullSecrets": [],
    "imageRegistry": "",
    "postgresql": {
        "auth": {
            "database": "",
            "existingSecret": "",
            "password": "",
            "postgresPassword": "",
            "secretKeys": {
                "adminPasswordKey": "",
                "replicationPasswordKey": "",
                "userPasswordKey": ""
            },
            "username": ""
        },
        "service": {
            "ports": {
                "postgresql": ""
            }
        }
    },
    "storageClass": ""
}
```

This causes the error. Any idea how to resolve this?

muja commented:
That missing section is defined in the `postgres` chart's `values.yaml` file and when the invocation happens in the `upstream` chart, those values are not yet set/merged. I'm not a helm chart expert so I'm not sure what would be a good way to resolve this other than specifying empty objects in the values file:

```yaml
global:
  postgres:
    auth: {}
upstream:
  auth: {}
```

muja commented on charts/backstage/values.yaml:
```diff
@@ -108,8 +108,8 @@ upstream:
-            key: postgres-password
-            name: '{{- include "janus-idp.postgresql.secretName" . }}'
+            key: '{{- include "postgresql.adminPasswordKey" . }}'
+            name: '{{- include "postgresql.v1.secretName" . }}'
```
using this helper here changes the `name` property from `rhdh-chart-backstage` to `rhdh-chart-postgresql`. Is this a breaking change / problem?

coreydaley committed:
Merge branch 'main' into main

muja commented:
Any update on this?

rm3l commented:
> Any update on this?

@muja Sorry for the delay. We'll review this PR as soon as possible. Thanks.

rm3l committed:
Merge branch 'main' into main

rm3l committed:
Merge branch 'main' into main

Fortune-Ndlovu committed:
Merge branch 'main' into main

rm3l reviewed (commented):
Hi @muja ,
So sorry it took us this long to get back to this PR. Are you still interested in following up on this?

rm3l committed:
Merge branch 'main' into main

rm3l committed:
Merge branch 'main' into main

rm3l committed:
Merge branch 'main' into main

rm3l committed:
Merge remote-tracking branch 'upstream/main' into muja/main

# Conflicts:
#	charts/backstage/values.schema.json
#	charts/backstage/values.yaml

rm3l committed:
fix(backstage): consistent usage of postgresql.adminPasswordKey

Replace hard-coded `postgres-password` key in POSTGRESQL_ADMIN_PASSWORD
env vars with the new `rhdh.postgresql.adminPasswordKey` helper, so
overriding `upstream.postgresql.auth.secretKeys.adminPasswordKey` is
consistently respected by both the Backstage and PostgreSQL containers.

rm3l committed:
Merge remote-tracking branch 'upstream/main' into muja/main
